### PR TITLE
feat: Adds the ability to edit saved segments

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

- Ability to edit contact segment query and name.

Fixes https://linear.app/chatwoot/issue/CW-2003/ability-to-edit-saved-segments

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Contact segment**
https://www.loom.com/share/7396fb18fddb42bca1f6a4681f07cc01

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by iamsivin** [comment]
Only merge this PR after merging [#7236 Adds the ability to edit saved folders](https://github.com/chatwoot/chatwoot/pull/7236).

**Review by scmmishra — APPROVED** [review]

**Pull request merged**
